### PR TITLE
Move trusted code into dedicated directory

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
In order to support both the untrusted and trusted interfaces to SGX the
trusted code as been moved into a dedicated trusted directory

### CODE REVIEW

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: 2
 updates:
 - package-ecosystem: cargo
-  directory: "/"
+  directory: "/trusted"
   schedule:
     interval: daily
   open-pull-requests-limit: 10
```
Comment (remoun) [inline .github/dependabot.yaml:7]: Dependabot can also notify us of GHA updates 
```suggestion
  open-pull-requests-limit: 10
- package-ecosystem: "github-actions"
  directory: "/"
  schedule:
    # Check for updates to GitHub Actions every weekday
    interval: "daily"
```

File: .github/workflows/workspace.yaml [hunk 2]
```diff
@@ -22,15 +22,16 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: fmt
-          args: --all -- --check
+          args: --manifest-path trusted/Cargo.toml --all -- --check
```
Comment (remoun) [inline .github/workflows/workspace.yaml:25]: Why not use the working dir override?
```suggestion
          args: --all -- --check
        working-directory: ./trusted
```

Comment (nick-mobilecoin) [reply .github/workflows/workspace.yaml:25]: I wanted to :(, https://github.com/actions-rs/cargo/issues/6

Comment (remoun) [reply .github/workflows/workspace.yaml:25]: Fair enough. Seems worth documenting here

File: .github/workflows/panic-abort.yaml [hunk 3]
```diff
@@ -24,7 +24,7 @@ jobs:
       target:
         - x86_64-unknown-linux-gnu
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v3
       - uses: actions-rs/toolchain@v1
         with:
           toolchain: ${{ matrix.rust }}
```
[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply .github/workflows/panic-abort.yaml:31]: added now

File: .github/workflows/workspace.yaml (hunk 2 above)
Comment (nick-mobilecoin) [reply .github/workflows/workspace.yaml:27]: added a comment

File: .github/workflows/panic-abort.yaml [hunk 4]
```diff
@@ -3,12 +3,12 @@ name: panic-abort
 on:
   pull_request:
     paths:
-      - "panic/abort/**"
+      - "trusted/panic/abort/**"
```
[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply .github/workflows/panic-abort.yaml]: renamed file and made generic over trusted

**Review by remoun — APPROVED** [review]

**Review by wjuan-mob — APPROVED** [review]

**Pull request merged**